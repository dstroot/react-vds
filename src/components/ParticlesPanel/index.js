import React from 'react';
import Particles from 'react-particles-js';
import ErrorBoundary from 'ErrorBoundary';

// particles configuration
// https://vincentgarreau.com/particles.js/
import config from './particles-config.json';

const ParticlesPanel = () => (
  <ErrorBoundary>
    <Particles params={config} className={'canvas-container'} />
  </ErrorBoundary>
);

export default ParticlesPanel;
