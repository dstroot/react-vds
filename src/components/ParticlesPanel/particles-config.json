{
  "particles": {
    "number": {
      "value": 30,
      "density": {
        "enable": true,
        "value_area": 100
      }
    },
    "color": {
      "value": "#7256ad"
    },
    "shape": {
      "type": "circle",
      "stroke": {
        "width": 0,
        "color": "#7256ad"
      }
    },
    "size": {
      "value": 3,
      "random": true
    },
    "line_linked": {
      "enable": true,
      "distance": 110,
      "color": "#7256ad",
      "opacity": 1,
      "width": 1
    },
    "move": {
      "enable": true,
      "speed": 3
    }
  },
  "retina_detect": true
}
