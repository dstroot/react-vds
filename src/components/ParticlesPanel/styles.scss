.outer {
  position: relative;
  max-width: 100%;
  max-height: 100%;
  margin: 0 auto;
  padding: 0;
}

.canvas-container {
  height: 75%;
}
