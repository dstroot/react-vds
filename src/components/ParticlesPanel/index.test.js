import React from 'react';
import renderer from 'react-test-renderer';

import ParticlesPanel from '../ParticlesPanel';

test('ParticlesPanel renders', () => {
  const component = renderer.create(<ParticlesPanel />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
