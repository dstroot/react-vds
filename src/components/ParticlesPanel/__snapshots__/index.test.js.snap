// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParticlesPanel renders 1`] = `
<div
  className="canvas-container"
  id="tsparticles"
>
  <canvas
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  />
</div>
`;
